File: backend/internal/handler/dto/message.go
package dto

import "time"

// MessageResponse is the API response DTO for trip messages.
type MessageResponse struct {
	ID          string            `json:"id"`
	TripID      string            `json:"trip_id"`
	ParentID    *string           `json:"parent_id,omitempty"`
	UserID      string            `json:"user_id"`
	DisplayName string            `json:"display_name"`
	Avatar      string            `json:"avatar"`
	Category    string            `json:"category"`
	Content     string            `json:"content"`
	Timestamp   time.Time         `json:"timestamp"`
	Replies     []MessageResponse `json:"replies"`
	CreatedAt   time.Time         `json:"created_at"`
	CreatedBy   string            `json:"created_by"`
	UpdatedAt   time.Time         `json:"updated_at"`
	UpdatedBy   string            `json:"updated_by"`
}

File: app/lib/domain/entities/message.dart
import 'package:freezed_annotation/freezed_annotation.dart';

part 'message.freezed.dart';
part 'message.g.dart';

/// 留言領域實體 (Domain Entity)
///
/// 不可變物件，承載留言的業務邏輯。
@freezed
abstract class Message with _$Message {
  const Message._();

  const factory Message({
    required String id,
    String? tripId,
    String? parentId,
    @Default('') String userId,
    @Default('') String user,
    @Default('🐻') String avatar,
    @Default('') String category,
    @Default('') String content,
    required DateTime timestamp,
    required DateTime createdAt,
    required String createdBy,
    required DateTime updatedAt,
    required String updatedBy,
  }) = _Message;

  /// 是否為回覆留言
  bool get isReply => parentId != null;

  factory Message.fromJson(Map<String, dynamic> json) => _$MessageFromJson(json);
}
